app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -837,6 +828,13 @@ export default Ember.Component.extend({
 
         this.appEvents.trigger("composer:insert-text", markdown);
         handled = true;
+      } else if (plainText) {
```

Comment:
This is not intended behavior. I pushed the commit in https://github.com/discourse/discourse/commit/dc2ee78f7a667cd0ded8e198282def7ea51797bb.

Reply:
@vinothkannans can you tell me why this isn't intended behavior? It's always better if I know what exactly I did wrong.

Reply:
This `else if` condition will run only if ["HTML pasting"](https://github.com/venarius/discourse/blob/044a15afcb94e9a359fd95cb3ef82672cdf6c515/app/assets/javascripts/discourse/components/d-editor.js.es6#L823-L830) not happen. The intended behavior is table detection should happen even before HTML pasting. Only if pasting is not inside code block and HTML pasting enabled.

Reply:
Oh okay, got it! Thanks!